tests/suites/test_suite_gcm.function

```diff
@@ -16,15 +16,19 @@ static int check_multipart( mbedtls_gcm_context *ctx,
     int ok = 0;
     uint8_t *output = NULL;
     size_t n2 = input->len - n1;
+    size_t n1_add = add->len/2;
```

Comment:
since n1 is incremented in a loop and not related to n1_add, I decided to just split the associated data in half.

Reply:
Splitting in half is not a good test because the test data lengths don't vary much. The reason `n1` is incremented in a loop is that it gives good test coverage: at the very least, we want the split to happen at 16*a+b for all b=0..15, because there's a risk of bugs related to how far the split is into a 16-byte block.

Correlating `n1` with `n1_add` is not ideal but better than `add->len/2`. Actually I think `add->len - n1` would give slightly better coverage (keeping it within the range `0`..`add->len` of course).